### ISSUE TITLE
Max. allowed repetitions

### ISSUE THREAD

**Issue by RobertSchaefer** [foex-open-source/fos-progress-bar#2]
Great Plugin!! 
There's an issue with the max. allowed repetitions, which is limited to 50 (line 52 in script.js). Only 50 allowed refreshes might be not enough for long-running processes. This limitation cannot be increased by the plugin attribute "Number of Repetitions" either.

Best regards,
Robert

**Comment by foex-open-source** [comment]
Hi, thanks for opening the issue!
It's not documented, but that value can be overridden by using the **config.lastEndInterval** in the _"JavaScrip Initialization Code"_ attribute.
```
function(config){
    config.lastEndInterval = 100;
    return config;
}
```
The help text will be updated!

Btw, while investigating this issue, we just found another one #3.

**Comment by RobertSchaefer** [comment]
Hi, thanks for your fast reply!
Yes, overriding the value via "JavaScrip Initialization Code" will work. But this parameter is already covered by the plugin attribute "Number of Repetitions". So the developer's natural approach would be to use the plugin attribute. And there are all values higher than 50 ignored.

Best regards,
Robert

**Comment by foex-open-source** [comment]
The sole purpose of this value is to prevent "infinite intervals",  to ensure that the timer is removed at some point.

In a case when the _"Repetition"_ attribute is set to _"Until Progress is Complete"_  - and due to some issue on the server the progress will never complete - the plug-in would keep sending out the requests, without an end. 

But yes, you're right, it doesn't really make sense if the _"Repetition"_ is set to _"Maximum Number of Repetitions
"_ ; the _"Number of Repetitions"_ should be the last-stop value. 

Thanks for pointing it out!

**Comment by foex-open-source** [comment]
Fixed with v22.1.0
